pkg/parse/run.go

```diff
@@ -185,7 +188,8 @@ func Run(ctx context.Context, p Parser) {
 func run(ctx context.Context, p Parser, trigger string, state *reconcilerState) {
```

Comment:
`run` is used by several different timers - `resyncTimer`, `runTimer`, `retryTimer`, etc. I'm curious if we want to use this same 10 minute retry logic for all of these entrypoints? It seems this could delay re-entry into the loop so that the next time can be reached. WDYT?

As a side note - this function reaches the retry logic for both `SourceCommitAndDir` and `readHydratedDir`. In the unlikely worst case, that means we could spend ~20 minutes in this function doing retries.

Reply:
`run` is used by all the timers. Theoretically, `resyncTimer` and `retryTimer` should use a shorter timeout than `runTimer` because the source and hydratedDir should already be computed. However, the `run` function with all the timers re-reads the file system, which means the source can change at any time. I think using the same retry logic guarantees the source and hydratedDir can be recomputed.

With a 5 mins retry timeout for `SourceCommitAndDir` and 1 min for `readHydratedDir`, it adds up to 6 mins in the worst case, which seems to be okay.

Reply:
changed to 5 mins and 1 min